### ISSUE TITLE
does the variadic arg must have at least one? or it's a bug

### ISSUE THREAD

**Issue by lydiandy** [vlang/v#3718]
**V version:**  
latest: V 0.1.25 d9cf98f
**OS:** 
macos
**What did you do?**
the follow sample code should be fine:
```v
fn sum_int(init int,val ...int) int {
	mut sum:=init
	for i in val {
		sum+=i
	}
	return sum
}

fn main(){
	ii:=sum_int(1,2,3,4) //this line is passed
	ii:=sum_int(1) //this line report a panic
	println(ii)
}
```

**What did you expect to see?**

**What did you see instead?**
```
V panic: array.get: index out of range (i == 0, a.len == 0)
0   v                                   0x000000010a583340 v_panic + 48
1   v                                   0x000000010a583908 array_get + 88
2   v                                   0x000000010a610405 compiler__Parser_fn_call_vargs + 1813
3   v                                   0x000000010a5e71a3 compiler__Parser_fn_call_args + 10227
4   v                                   0x000000010a5e3d2b compiler__Parser_fn_call + 2315
5   v                                   0x000000010a608cd1 compiler__Parser_name_expr + 7585
6   v                                   0x000000010a60cd9f compiler__Parser_factor + 2415
7   v                                   0x000000010a60c37c compiler__Parser_unary + 396
8   v                                   0x000000010a5f1169 compiler__Parser_term + 57
9   v                                   0x000000010a5f0c2b compiler__Parser_indot_expr + 43
10  v                                   0x000000010a5ce449 compiler__Parser_expression + 57
11  v                                   0x000000010a605bcb compiler__Parser_bterm + 43
12  v                                   0x000000010a5d68b2 compiler__Parser_bool_expression + 82
13  v                                   0x000000010a5df5e0 compiler__Parser_gen_var_decl + 48
14  v                                   0x000000010a5d5a26 compiler__Parser_var_decl + 1942
15  v                                   0x000000010a5ccc2e compiler__Parser_statement + 446
16  v                                   0x000000010a5bee46 compiler__Parser_statements_no_rcbr + 118
17  v                                   0x000000010a5c4311 compiler__Parser_fn_decl + 11761
18  v                                   0x000000010a5bf8fb compiler__Parser_parse + 2523
19  v                                   0x000000010a616101 compiler__V_parse + 305
20  v                                   0x000000010a616e9b compiler__V_compile + 3403
21  v                                   0x000000010a627d21 main__compile + 193
22  v                                   0x000000010a62c009 main__main + 1545
23  v                                   0x000000010a62e65e main + 78
24  libdyld.dylib                       0x00007fff71a687fd start + 1
```

**Comment by changrui** [comment]
It may be the following:
```
fn sum(val ...int) int {
  mut s := 0
   for i in val {
      s += i
  }
  return s
}
```

calling the function `sum_int(1)`, at least ONE argument.

**Comment by lydiandy** [comment]
yeah, it is also ok
but I think  allow no argument is better,just return an empty array

**Comment by joe-conigliaro** [comment]
there is many bugs with current variadic implementation that are not getting fixed, they will be fixed in the new compiler

**Comment by alexesprit** [comment]
Fixed in 9c6ac7cb71f9227d74e75b08458f0ff747770317 and #4229.
